src/app/model/Rock.ts

```diff
@@ -0,0 +1,109 @@
+import { Board } from './Board';
+import { PieceMovement } from './PieceMovement';
+import { Side, Square } from './Types';
+
+export class Rock implements PieceMovement {
+  id: string;
+  side: Side;
+
+  constructor(id: string, side: Side) {
+    this.id = id;
+    this.side = side;
+  }
+
+  possibleMoves(position: Square, board: Board): Square[] {
+    let movesToGo: Square[] = [];
+    movesToGo.concat(this._goUp(position, board));
+    movesToGo.concat(this._goDown(position, board));
+    movesToGo.concat(this._goLeft(position, board));
+    movesToGo.concat(this._goRight(position, board));
+
+    return movesToGo;
+  }
+
+  _goUp(position: Square, board: Board): Square[] {
+    let movesToGo: Square[] = [];
+    let matrixDimension = board.squares[0].length;
+
+    if (position.row === 8) {
+      return movesToGo;
+    } else {
+      for (let i = position.row; i < matrixDimension; i++) {
+        if (board.squares[i][position.column - 1].piece == null) {
+          movesToGo.push(board.squares[i][position.column - 1]);
+        } else {
+          if (board.squares[i][position.column - 1].piece?.side !== this.side) {
+            movesToGo.push(board.squares[i][position.column - 1]);
+            break;
+          } else break;
+        }
+      }
+    }
+
+    return movesToGo;
+  }
+
+  _goDown(position: Square, board: Board): Square[] {
+    let movesToGo: Square[] = [];
+
+    if (position.row === 1) {
+      return movesToGo;
+    } else {
+      for (let i = position.row - 2; i >= 0; i--) {
+        if (board.squares[i][position.column - 1].piece == null) {
+          movesToGo.push(board.squares[i][position.column - 1]);
+        } else {
+          if (board.squares[i][position.column - 1].piece?.side !== this.side) {
```

Comment:
Dałbyś radę ten warunek jakoś odpowiednio opisać i wyciągnąć go do metody, która mówi o co chodzi?
Piece to nie musi być type. Może coś Ci się przyda poza samymi polami? Np. jakieś metody zrobią to bardziej czytelne :)

Reply:
if wyciągnięty do metody :-)

co do Piece:
no chciałem od pocżątku Piece jako klasę i wtedy by każdy pionek bo niej dziedziczył przy okazji, ale mówiłeś, żebym zostawił dziedziczenie póki co :-p

Reply:
Tzn. to chyba było na etapie samej wieży, tak mi się wydaje - albo chodziło o skupienie się na czymś innym.
Albo może po prostu późno było :P
Ale dobra, to spróbuj i zobaczmy jak to będzie :)